### PR TITLE
VACMS-16457: Update/patch entity_browser module

### LINKED ISSUE #16457: Update/patch entity_browser module
## User Story or Problem Statement
The recent update to the `entity_browser` module, v2.10.0, contains some breaking changes. I am not entirely sure what the cause of all of the issues are. These issues were identified in cypress testing. The content types that failed are:
- step_by_step
- campaign_landing_page
- checklist
- q_a
- support_resources_detail_page
See https://pr16385-be6rnqikyapzpk5k5fqytdpcocri6pxn.ci.cms.va.gov/cypress/ for more details. 

## Description or Additional Context 
See https://www.drupal.org/project/entity_browser/issues/3408216#comment-15362842

The error is: 
```
Error: Call to a member function getStorage() on null in Drupal\entity_browser\Plugin\Field\FieldWidget\EntityReferenceBrowserWidget->formElementEntities() (line 759 of modules/contrib/entity_browser/src/Plugin/Field/FieldWidget/EntityReferenceBrowserWidget.php).
Drupal\entity_browser\Plugin\Field\FieldWidget\EntityReferenceBrowserWidget->formElement(Object, 0, Array, Array, Object) (Line: 130)
Drupal\entity_browser_table\Plugin\Field\FieldWidget\EntityReferenceBrowserTableWidget->formElement(Object, 0, Array, Array, Object) (Line: 342)
Drupal\Core\Field\WidgetBase->formSingleElement(Object, 0, Array, Array, Object) (Line: 92)
Drupal\Core\Field\WidgetBase->form(Object, Array, Object) (Line: 183)
Drupal\Core\Entity\Entity\EntityFormDisplay->buildForm(Object, Array, Object) (Line: 669)
Drupal\paragraphs\Plugin\Field\FieldWidget\InlineParagraphsWidget->formElement(Object, 0, Array, Array, Object) (Line: 342)
Drupal\Core\Field\WidgetBase->formSingleElement(Object, 0, Array, Array, Object) (Line: 849)
Drupal\paragraphs\Plugin\Field\FieldWidget\InlineParagraphsWidget->formMultipleElements(Object, Array, Object) (Line: 111)
Drupal\Core\Field\WidgetBase->form(Object, Array, Object, NULL) (Line: 968)
Drupal\paragraphs\Plugin\Field\FieldWidget\InlineParagraphsWidget->form(Object, Array, Object) (Line: 183)
Drupal\Core\Entity\Entity\EntityFormDisplay->buildForm(Object, Array, Object) (Line: 121)
Drupal\Core\Entity\ContentEntityForm->form(Array, Object) (Line: 127)
Drupal\node\NodeForm->form(Array, Object) (Line: 107)
Drupal\Core\Entity\EntityForm->buildForm(Array, Object)
call_user_func_array(Array, Array) (Line: 536)
Drupal\Core\Form\FormBuilder->retrieveForm('node_step_by_step_form', Object) (Line: 283)
Drupal\Core\Form\FormBuilder->buildForm(Object, Object) (Line: 73)
Drupal\Core\Controller\FormController->getContentResult(Object, Object) (Line: 39)
Drupal\layout_builder\Controller\LayoutBuilderHtmlEntityFormController->getContentResult(Object, Object)
call_user_func_array(Array, Array) (Line: 123)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 592)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 124)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array) (Line: 97)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 181)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 76)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 68)
Drupal\simple_oauth\HttpMiddleware\BasicAuthSwap->handle(Object, 1, 1) (Line: 58)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 106)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 85)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 53)
Asm89\Stack\Cors->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 51)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 51)
Drupal\Core\StackMiddleware\StackedHttpKernel->handle(Object, 1, 1) (Line: 704)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)
```

## Acceptance Criteria
- [ ] Patch `entity_browser` module so that it functions correctly
- [ ] Ensure the `entity_browser_table` module is functioning correctly, patch if necessary
- [ ] All automated tests must pass. 

## Team
Please check the team(s) that will do this work.

- [ ] CMS Team
- [ ] Public Websites
- [ ] Facilities
- [ ] Accelerated Publishing

### PR DESCRIPTION
**Pull request by edmund-dunn**
## Description

Closes #16457.

This updates the `entity_browser` module and patches it and the `entity_browser_tables` module. There were breaking changes in `entity_browser` see https://www.drupal.org/project/entity_browser/issues/3407702 for reference.

## Testing done
Tested locally. Cypress tests that failed in the dependabot PR passed locally. 

## QA steps

What needs to be checked to prove this works?
What needs to be checked to prove it didn't break any related things?
What variations of circumstances (users, actions, values) need to be checked?

Mostly this would be considered good to go if autoated tests pass. However in light of the information on breaking changes we need each product team to review to make sure that none of their custom code is effected by those breaking changes.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [x] `CMS Team`
- [x] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by dsasser** [comment]
@edmund-dunn changes look good but there appears to be a cypress test failure.

**Comment by edmund-dunn** [comment]
> @edmund-dunn changes look good but there appears to be a cypress test failure.

@dsasser that test is no longer needed. Per https://www.drupal.org/project/entity_browser/issues/3074457 that is now part of entity_browser.

**Comment by BerniXiongA6** [comment]
@edmund-dunn would you be able to size this one since i'ts already in progress? Thank you very much cc: @maortiz-27-80

**Review by ndouglas — APPROVED** [review]
This is fine to merge from my perspective once merge conflicts are resolved 🙂

**Review by dsasser — COMMENTED** [review]
The patch for entity_browser is a bit weird in that it is loading entities that ostensibly should already be there. Swirt had a similar sentiment [on the d.o issue](https://www.drupal.org/project/entity_browser/issues/3408216#comment-15379911).

**Comment by edmund-dunn** [comment]
@dsasser that was a good call. I was able to remove that patch. The patch on entity_borwser_tabloe to bring it in line with the changes to entity_browser seemed to have fixed our issues.

File: patches/3408216-error-call-to-member-function-getstorage-on-null.patch
Comment (dsasser) [inline patches/3408216-error-call-to-member-function-getstorage-on-null.patch:1]: SHOULD:

Do we need this patch if we are no longer including it in Composer?

**Review by dsasser — CHANGES_REQUESTED** [review]

File: patches/3408216-error-call-to-member-function-getstorage-on-null.patch
Comment (edmund-dunn) [reply patches/3408216-error-call-to-member-function-getstorage-on-null.patch:1]: It is removed. Good cal!

**Review by dsasser — APPROVED** [review]

**Review by swirtSJW — APPROVED** [review]
I am happy that original patch has been removed.

### QA Steps
Review instances of entity browser:
image
There are only 2 instances of entity_browser_table being used on Facilities products do my QA is limited to those.   In the absence of actual QA steps on the PR, I am assuming that CMS team and PW team have reviewed the other instances. 

- [x] Validate that vet_center_locations_list is working as it should be
prod on right
image

added entities successfully
image

Removed entities successfully
image


- [x] Validate that paragraph.contact_information  is working as it should be
Prod on right
image

Added and removed phone numbers fine.  Allowed number of entries still work
image

**Pull request merged**